src/app/state/operations/tokens.ts

```diff
@@ -0,0 +1,63 @@
+import { Dispatch } from 'react'
+import Eth from 'ethjs'
+import erc720 from './erc720.json'
+import { getAccounts, getNetwork } from '../../../ethrpc'
+import { addToken, addTokenData } from '../reducers/tokens'
+import { getTokens } from '../../../config/getConfig'
+import { saveToLocalStorage, getValueFromLocalStorage } from '../../../features/localStorage'
+
+export const getTokenList = (provider: any) => (dispatch: Dispatch<any>) =>
+  getNetwork(provider)
+    .then((chainId: string) => {
+      const tokens = getTokens(parseInt(chainId))
+      const localTokens = getValueFromLocalStorage(chainId, 'TOKEN')
+      const all = tokens.concat(localTokens)
+      Array.isArray(all) && all.map((address: string) => {
+        dispatch(getToken(provider, address))
+      })
+    })
+
+export const getToken = (provider: any, address: string, onComplete?: any) => (dispatch: Dispatch<any>) => {
+  const eth = new Eth(provider)
+  const token = eth.contract(erc720).at(address)
+
+  eth.getCode(address)
+    .then((result: string) => {
+      if (result === '0x0') {
+        return onComplete ? onComplete(new Error('Not a contract')) : null
+      }
+
+      dispatch(addToken({ address }))
+
+      token.symbol().then((symbol: string[]) => symbol[0])
+        .then((symbol: string) => dispatch(addTokenData({ data: { address, symbol } }))).catch()
```

Comment:
In a ERC20 and 721 token contract, `name` and `symbol` are optional functions. So, if they don't exist in the contract, it will throw an error here. I don't want it to break, I want to continue to the next call, so I add the catch() here.

Reply:
Should we catch ONLY the error thrown by this case? Instead of catching everything. Maybe it throws another error...

Reply:
The problem is that the errors returned by ethjs are strings not objects. This a similar issue to the previous PR where we tried to catch the error. Here is the error that is returned:
```
[ethjs-query] while formatting outputs from RPC '{"value":{"code":-32603,"message":"Non-200 status code: '500'","data":{"jsonrpc":"2.0","id":3846821291,"error":{"code":-32015,"message":"VM execution error: transaction reverted"}}}}'
```